### ISSUE TITLE
Watch together feature

### ISSUE THREAD

**Issue by odmivolsi** [jaruba/PowderWeb#46]
I read you have an option to sync playback if multiple people login with the same account but most people aren't willing to share their account details even to their friends. Is there I can let people with separate accounts watch together by entering their account ID or something like that? This will be a great feature and will enhance the viewing experience for people. Thanks :)

**Comment by odmivolsi** [comment]
@jaruba help me here man

**Comment by jaruba** [comment]
Sorry, I was on vacation, as the description says, u can sync playback with others only if you are using the same account. (even this is quite limited for now)

Although there is no direct feature of this app that allows shared viewing, there are other apps that can be used for this purpose and they work very well, my personal favorite for this feature is Parsec, it's free, it will let u share ur screen and controls with others and also has a screen chat feature. Another alternative solution might be using the Metastream browser extension, although I believe this may be more complicated to set up for more people.

**Comment by odmivolsi** [comment]
Thanks man. Is there any way to integrate parsec into the player as most viewers that are going to visit my site aren't going to be good with computers and installing and setting up additional extensions will be hard for them

**Comment by jaruba** [comment]
Parsec is a separate program, so unfortunately not. It may be possible to integrate Metastream, which would require installing a browser extension, this might be pretty straight forward for users..

The best way to do this would be to just integrate a synced playback feature in Powder Web directly, I like the idea of such a feature, as Powder Web is pretty mature now in terms of other features. But it may take a long time to make it as my time is limited these days and it's a hobby project for me.

**Comment by odmivolsi** [comment]
Yes a synced playback feature will be AMAZING! Please try to do it or guide me on how i can implement it on my own.
Thanks a ton!

**Comment by jaruba** [comment]
What if this feature would allow a user to share the video with anyone they share a link to? (even if those people would not have direct access to the server themselves)

This would be doable, but the issue here is that depending on what video/audio encoding the original video has, and depending on what devices each of the viewer has, it may use up much more resources (to transcode the video) from the server's side for each viewer.

The streaming link would only be valid on a per video basis though, and could also be made to expire after some time.

But, from an implementation point of view, it would be easier to make a feature that would sync playback with anyone you give a link to then a feature that would sync playback only between users that have accounts to the server. It would also end in a better feature all together, but the issue of server resource usage and security might be a problem, as you could theoretically share a video to someone that does not have access, and allow them to view (even temporarily) without requiring an account.

I'm curious on your point of view regarding this.

**Comment by odmivolsi** [comment]
Hey @jaruba ! I was caught up with work and just now saw your comment.

This sounds amazing honestly, sending a link for viewing can work wonderfully as well and you could implement it in two ways - 

1. If I am sending a link to another person who has an account, the person can get a notification saying "XYZ wants to watch a movie with you" apart from normal ways of sending the link via whatsapp, gmail, etc.

2. For the part that a person without an account can also be viewing- we can let anyone view if they have a link but to create a link you MUST have an account. This will encourage sign-ups as well.

Let me know what you think about this :)

**Comment by jaruba** [comment]
The reason I asked about sharing to non-members is because users shouldn't have access to the other user accounts, or have any way of discovering others on the system. As that would be a privacy concern.

While allowing others to just share links to anyone (if allowed by the administrator) would be significantly easier as embedding is already supported. To ensure that other websites don't just embed these shared viewing pages and thus steal resources from the server, the shared viewing links could expire after a certain time. (a few hours or a day)

**Comment by jaruba** [comment]
I would avoid the 1st point of your suggestions though, for the same reason as I wouldn't allow sharing between accounts: privacy.

**Comment by odmivolsi** [comment]
@jaruba sending the links can work just fine imo. But it will be nice to have an option for point 1 in case the admin wants to create a mini social network for people to watch stuff together.
Also, i dont understand how it can be a privacy concern. Mind explaining? Thanks :)

**Comment by jaruba** [comment]
> a mini social network for people to watch stuff together.

Creating a social network is outside the scope of this project. But it has an API, so web developers can build social networks on top of it if they want, but social networking should not be the target of this project at all.

> i dont understand how it can be a privacy concern

Privacy presumes that user information is not shared to others, at all. You mentioned showing a message with "XYZ wants to watch a movie with you", in this case a user would be notified that another user called "XYZ" exists, which is a privacy issue.

**Comment by odmivolsi** [comment]
> so web developers can build social networks on top of it if they want

Yes i was just giving a hypothetical situation. The link sharing will be a great feature. Hope to see it in action soon :)

> Privacy presumes that user information is not shared to others, at all. You mentioned showing a message with "XYZ wants to watch a movie with you", in this case a user would be notified that another user called "XYZ" exists, which is a privacy issue.

Got it now

**Comment by odmivolsi** [comment]
Hey @jaruba ! Did you get a chance to work on this? I came across metastream which looks great to me. Have a look, you might like it https://github.com/samuelmaddock/metastream

**Comment by jaruba** [comment]
I didn't get a chance for it, no.. I also suggested Metastream to you in this comment some time ago: https://github.com/jaruba/PowderWeb/issues/46#issuecomment-657697524

**Comment by odmivolsi** [comment]
Ahh I see and sorry, it slipped my clumsy head :D

This feature would be a really nice addition, please try to work on it in case you get time to

**Comment by odmivolsi** [comment]
Hey @jaruba ! Did you get a chance to work on this? Im really excited for this feature
